src/internal/m365/collection/groups/collection.go

```diff
@@ -268,10 +261,13 @@ func (col *Collection) streamItems(ctx context.Context, errs *fault.Bus) {
 			writer := kjson.NewJsonSerializationWriter()
 			defer writer.Close()
 
+			flds := col.fullPath.Folders()
+			parentFolderID := flds[len(flds)-1]
+
 			item, info, err := col.getter.getChannelMessage(
 				ctx,
 				col.protectedResource,
-				col.chanID,
```

Comment:
I'd actually prefer a separate variable for the channel ID rather than extracting it from the storage path. Taking it from the storage path implies we'll never change the storage path format (I hope we don't, but best to avoid the coupling if we can)

Reply:
This doesn't prevent us from changing the storage path.  It just means that if/when we do, we'll need to field the channelID some other way to support that refactor.  While I understand the want to avoid copuling, I think this is fairly harmless in the scheme of things.